src/transformers/models/patchtsmixer/modeling_patchtsmixer.py

```diff
@@ -0,0 +1,2190 @@
+                seed_number=config.seed_number,
+            )
+        else:
+            self.masking = None
+
+        if config.scaling == "mean" or config.scaling is True:
+            self.scaler = PatchTSMixerMeanScaler(dim=1, keepdim=True)
+        elif config.scaling == "std":
+            self.scaler = PatchTSMixerStdScaler(dim=1, keepdim=True)
+        else:
+            self.scaler = PatchTSMixerNOPScaler(dim=1, keepdim=True)
+
+        # Initialize weights and apply final processing
+        if config.post_init:
+            self.post_init()
+
+    @replace_return_docstrings(output_type=PatchTSMixerModelOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        past_values: torch.Tensor,
+        observed_mask: Optional[torch.Tensor] = None,
+        output_hidden_states: Optional[bool] = False,
+        return_dict: Optional[bool] = None,
+    ) -> PatchTSMixerModelOutput:
+        r"""
+        Args:
+            past_values (`torch.FloatTensor` of shape `(batch_size, seq_length, input_size)`):
+                Context values of the time series. For a pretraining task, this denotes the input time series to
+                predict the masked portion. For a forecasting task, this denotes the history/past time series values.
+                Similarly, for classification or regression tasks, it denotes the appropriate context values of the
+                time series.
+
+                For univariate time series, `input_size` dimension should be 1. For multivariate time series, it is
+                greater than 1.
+            observed_mask (`torch.FloatTensor` of shape `(batch_size, sequence_length, input_size)`, *optional*):
+                Boolean mask to indicate which `past_values` were observed and which were missing. Mask values selected
+                in `[0, 1]`:
+                    - 1 for values that are **observed**,
+                    - 0 for values that are **missing** (i.e. NaNs that were replaced by zeros).
+
+            output_hidden_states (`bool`, *optional*):
+                Whether or not to return the hidden states of all layers.
+            return_dict (`bool`, *optional*):
+                Return dict.
+
+        Returns:
+            [`PatchTSMixerModelOutput`]
+        """
+
+        mask = None
+        if observed_mask is None:
+            observed_mask = torch.ones_like(past_values)
+        scaled_past_values, loc, scale = self.scaler(past_values, observed_mask)
+
+        patched_x = self.patching(scaled_past_values)  # [batch_size x input_size x num_patch x patch_len
+
+        enc_input = patched_x
+        if self.masking is not None:
+            enc_input, mask = self.masking(patched_x)
+            # enc_input: [batch_size x input_size x num_patch x patch_len]
+            # mask: [batch_size x input_size x num_patch]
+
+        encoder_output = self.encoder(enc_input, output_hidden_states=output_hidden_states)
+
+        return PatchTSMixerModelOutput(
```

Comment:
Can we add a `if return_dict: ...` else statement here?

Reply:
https://github.com/huggingface/transformers/pull/26247/commits/3daac8886a75ab9c285c172c5bd5e6073973c4aa